app/javascript/client/checkout/utils.tsx

```diff
@@ -34,33 +34,42 @@ export const validationSchema = yup.object({
         .required()
         .email()
         .label('Email'),
+    taxExempt: yup.boolean(),
+    sameAddress: yup.boolean(),
     shippingName: yup
         .string()
-        .required()
-        .label('Name'),
+        .when('sameAddress', {
```

Comment:
I re-did these validations for the `shippingAddress` fields to make them conditional upon `sameAddress` being false and it seems to work now